Gemfile

```diff
@@ -1,17 +1,18 @@
 source "http://rubygems.org"
 
-# Declare your gem's dependencies in bootstrap_form.gemspec.
-# Bundler will treat runtime dependencies like base dependencies, and
-# development dependencies will be added by default to the :development group.
 gemspec
 
-# jquery-rails is used by the dummy application
-gem "jquery-rails"
+# Uncomment and change rails version for testing purposes
+# gem "rails", "~> 5.2.0.beta2"
 
-# Declare any dependencies that are still in development here instead of in
-# your gemspec. These might include edge Rails or gems from your path or
-# Git. Remember to move these dependencies to your gemspec before releasing
-# your gem to rubygems.org.
+group :development do
+  gem "htmlbeautifier"
+end
 
-# To use debugger
-# gem 'ruby-debug19', :require => 'ruby-debug'
+group :test do
+  gem "diffy"
+  gem "equivalent-xml"
+  gem "mocha"
+  gem "sqlite3"
+  gem "timecop", "~> 0.7.1"
```

Comment:
Could you explain further why these dependencies belong in the Gemfile and not the as dev dependencies in the gemspec? It means that we have to repeat these in each of the test gemfiles, and I don't see much benefit.

Reply:
This is mostly to keep gemspec clean. Unless you're directly working on the gem you don't need to install or even care about those dev dependencies. Gives you flexibility to cram whatever you want into your own Gemfile. See what Rails does: https://github.com/rails/rails/blob/master/rails.gemspec Million of gems used for development and testing are defined in Gemfile instead.

Reply:
Interesting, I had no idea that was a common practice! I am still confused because the definition of gemspec dev dependencies is that they ["aren't installed by default and aren't activated when a gem is required"](https://github.com/rubygems/guides/blob/543ae72b78678d4f9b978f7070635d553687ec38/specification-reference.md#add_development_dependency). In other words, they are not considered at all unless you are working directly on the gem, which is exactly the behavior we want. Other popular Ruby projects like [capybara](https://github.com/teamcapybara/capybara/blob/e0afe31a1e4881b1a3cf082caba1fb43f311592e/capybara.gemspec#L30-L41), [sprockets](https://github.com/rails/sprockets/blob/9d945fbe1342136edd80888eba1c22f3d99d4ff9/sprockets.gemspec#L17-L33), and [rack](https://github.com/rack/rack/blob/2e92a25342b5d51f4094366c3b8dd797cbd208a4/rack.gemspec#L31-L33) do it the gemspec way, so maybe it is just personal preference?

Reply:
BTW I don't mean to be argumentative or wish to hold up this PR in any way. I am honestly just really curious why the Ruby community seems to have two different ways of accomplishing the same thing. Maybe the Gemfile is preferable when you have to platform-specific dependencies, and in that case the Gemfile DSL is more expressive? Feel free to ignore my ramblings. 😀